static/js/publisher/pages/RegisterSnap/__tests__/RegisterSnap.test.tsx

```diff
@@ -0,0 +1,151 @@
+import { RecoilRoot } from "recoil";
+import { BrowserRouter } from "react-router-dom";
+import { QueryClient, QueryClientProvider } from "react-query";
+import { http, HttpResponse } from "msw";
+import { setupServer } from "msw/node";
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import "@testing-library/jest-dom";
+
+import RegisterSnap from "../RegisterSnap";
+
+const queryClient = new QueryClient();
+
+function renderComponent() {
+  return render(
+    <RecoilRoot>
+      <QueryClientProvider client={queryClient}>
+        <BrowserRouter>
+          <RegisterSnap />
+        </BrowserRouter>
+      </QueryClientProvider>
+    </RecoilRoot>,
+  );
+}
+
+const storesApiResponse = {
+  success: true,
+  data: [
+    {
+      id: "ubuntu",
+      name: "Global",
+    },
+    {
+      id: "test-store",
+      name: "Test store",
+    },
+  ],
+};
+
+const server = setupServer();
+
+beforeAll(() => {
+  server.listen();
+});
+
+afterEach(() => {
+  server.resetHandlers();
+});
+
+afterAll(() => {
+  server.close();
+});
+
+describe("RegisterSnap", () => {
+  test("selects global store as default", async () => {
+    server.use(
+      http.get("/api/stores", () => {
+        return HttpResponse.json(storesApiResponse);
+      }),
+    );
+
+    renderComponent();
+
+    await waitFor(() => {
+      expect(screen.getByLabelText("Store")).toHaveValue("ubuntu");
+    });
+  });
+
+  test("disables public and private inputs by default", async () => {
+    server.use(
+      http.get("/api/stores", () => {
+        return HttpResponse.json(storesApiResponse);
+      }),
+    );
+
+    renderComponent();
+
+    await waitFor(() => {
+      expect(screen.getByLabelText("Public")).toBeDisabled();
+      expect(screen.getByLabelText("Private")).toBeDisabled();
+    });
+  });
+
+  test("selects private by default", async () => {
+    server.use(
+      http.get("/api/stores", () => {
+        return HttpResponse.json(storesApiResponse);
+      }),
+    );
+
+    renderComponent();
+
+    await waitFor(() => {
+      expect(screen.getByLabelText("Private")).toBeChecked();
+    });
+  });
+
+  test("register button is enabled by default", async () => {
```

Comment:
I guess it was supposed to be "disabled"?

```suggestion
  test("register button is disabled by default", async () => {
```